chat-client-ui-types/uiContracts.ts

```diff
@@ -0,0 +1,92 @@
+import { InsertToCursorPositionParams } from '@aws/language-server-runtimes-types'
+export { InsertToCursorPositionParams } from '@aws/language-server-runtimes-types'
+
+export type AuthFollowUpType = 'full-auth' | 're-auth' | 'missing_scopes' | 'use-supported-auth'
+export function isValidAuthFollowUpType(value: string): value is AuthFollowUpType {
```

Comment:
Why are we exporting functions from ` types` package? Ideally `types` package is simply types without any side-effects and implementations.

Reply:
Ideally yes, but there is a strict dependency from ui types on the runtime types. The alternative is to define the same interface here which could cause issues in the future

Reply:
I was referring to `isValidAuthFollowUpType` function, but I guess it's fine to provide it as a helper type guard for consuming projects.